context/favourites.context.tsx

```diff
@@ -0,0 +1,60 @@
+'use client'
+
+import { createContext, useState, useEffect, useRef } from 'react'
+
+import { CatImage } from '@/model/app.model'
+
+type Props = {
+  children: React.ReactNode
+}
+
+type ContextType = {
+  favourites: CatImage[]
+  addToFavourites: (cat: CatImage) => void
+  removeFromFavourites: (cat: CatImage) => void
+}
+
+export const FavouritesContext = createContext<ContextType>({
+  favourites: [],
+  addToFavourites: () => {},
+  removeFromFavourites: () => {},
+})
+
+export const FavouritesProvider = ({ children }: Props) => {
+  const initialRender = useRef(true)
+  const [favourites, setFavourites] = useState<CatImage[]>([])
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const savedFavourites = window.localStorage.getItem('favourites')
+      setFavourites(savedFavourites ? JSON.parse(savedFavourites) : [])
+    }
+  }, [])
+
+  useEffect(() => {
+    if (initialRender.current) {

```

Comment:
Can you please describe what these 3-4 lines do?

Reply:
It is used to determine whether the current render is the initial render or a subsequent render.

Reply:
And how that does help the logic in this provider component in general?